src/notation/internal/notationparts.cpp

```diff
@@ -495,9 +501,15 @@ void NotationParts::appendLinkedStaff(Staff* staff, const ID& sourceStaffId, con
 {
     TRACEFUNC;
 
+    IF_ASSERT_FAILED(staff) {
+        return;
+    }
+
     Staff* sourceStaff = staffModifiable(sourceStaffId);
     Part* destinationPart = partModifiable(destinationPartId);
-    if (!staff || !sourceStaff || !destinationPart) {
+    if (!sourceStaff || !destinationPart) {
+        staff->unlink();
+        delete staff;
```

Comment:
I'm not a big fan of when a function deletes a passed argument (and it's not a data cleaning function like qDeleteAll). So a better solution would be to return bool from this function and delete the object at the level above

Reply:
No problemo, done and done. I wasn't sure if it was a wise idea to mess with the interface itself, but I guess the only two classes that implement it are the ones I'm working on, so I guess it's not a big deal